File: IQ/Classes/clsValidation.vb
Imports dataAccess

Public Class clsValidation

    Property ID As String
    Property description As String
    Property regEx As String
    Property ViolationMessage As String

    Public Sub New()

    End Sub

    Public Function Insert() As clsValidation
        Return New clsValidation(Me.description, Me.regEx, Me.ViolationMessage)
    End Function

    Public ReadOnly Property DisplayName(Language As clsLanguage)

        Get
            Return Me.description
        End Get

    End Property

    Public Sub New(id As Integer, description As String, regex As String, violation As String)

        Me.ID = id
        Me.description = description
        Me.regEx = regex
        Me.ViolationMessage = violation

        iq.Validations.Add(Me.ID, Me)

    End Sub

    Public Sub New(description As String, regex As String, violation As String)

        Me.description = description
        Me.regEx = regex
        Me.ViolationMessage = violation

        Dim sql$
        sql$ = "INSERT INTO [validation] (descripion,regex,violation) VALUES(" & da.SqlEncode(Me.description) & "," & da.SqlEncode(Me.regEx) & "," & da.SqlEncode(Me.ViolationMessage) & ");"
        Me.ID = da.DBExecutesql(sql$, True)

        iq.Validations.Add(Me.ID, Me)

    End Sub

    Public Sub update()

        Dim sql$
        sql$ = "UPDATE [validation] set "
        sql$ &= "description=" & da.SqlEncode(Me.description) & ","
        sql$ &= "regex=" & da.SqlEncode(Me.regEx) & ","
        sql$ &= "viloationmessage=" & da.SqlEncode(Me.ViolationMessage)

        da.dbexecutesql(sql$, False)

    End Sub




End Class

File: IQ/Classes/clsValidation.cs
using dataAccess;


public class clsValidation
{

    public string ID { get; set; }
    public string description { get; set; }
    public string regEx { get; set; }
    public string ViolationMessage { get; set; }

    public clsValidation()
    {

    }

    public clsValidation Insert()
    {
        return new clsValidation(this.description, this.regEx, this.ViolationMessage);
    }


    public dynamic get_DisplayName(clsLanguage Language)
    {
        return this.description;
    }


    public clsValidation(int id, string description, string regex, string violation)
    {

        this.ID = (id).ToString();
        this.description = description;
        this.regEx = regex;
        this.ViolationMessage = violation;

        iq.Validations.Add(int.Parse(this.ID), this);

    }

    public clsValidation(string description, string regex, string violation)
    {

        this.description = description;
        this.regEx = regex;
        this.ViolationMessage = violation;

        object sql = null;
        sql = "INSERT INTO [validation] (descripion,regex,violation) VALUES(" + da.SqlEncode(this.description) + "," + da.SqlEncode(this.regEx) + "," + da.SqlEncode(this.ViolationMessage) + ");";
        this.ID = (da.DBExecutesql(sql, true)).ToString();

        iq.Validations.Add(int.Parse(this.ID), this);

    }

    public void update()
    {

        object sql = null;
        sql = "UPDATE [validation] set ";
        sql += "description=" + da.SqlEncode(this.description) + ",";
        sql += "regex=" + da.SqlEncode(this.regEx) + ",";
        sql += "viloationmessage=" + da.SqlEncode(this.ViolationMessage);

        da.dbexecutesql(sql, false);

    }




}